src/hotspot/share/prims/unsafe.cpp

```diff
@@ -231,6 +231,7 @@ class MemoryAccess : StackObj {
 
   T get_volatile() {
     GuardUnsafeAccess guard(_thread);
+    assert(addr() != nullptr, "Attempting to load from null pointer");
```

Comment:
I don't see how `addr()` can be null unless `_obj` was null - which would be a usage error. So asserting `_obj != nullptr` in the constructor would seem better to me. I mean no point checking `addr()` here but not in other functions where we dereference it!